src/analysis/analysis-service.js

```diff
@@ -19,21 +19,48 @@ var AnalysisService = function (opts) {
   this._apiKey = opts.apiKey;
   this._authToken = opts.authToken;
   this._camshaftReference = opts.camshaftReference || camshaftReference; // For testing purposes
-  this._layersCollection = opts.layersCollection;
-  this._dataviewsCollection = opts.dataviewsCollection;
+  this._layersCollection = opts.layersCollection || [];
```

Comment:
IMO creating a new collection is a bad idea that could hide errors.

Anyway, it should be initialized with an empty collection not an array
```javascript
 this._layersCollection = opts.layersCollection || new Backbone.Collection([]);
```

Reply:
Ups this mustn't be there. I prefer without || [] instead of requiring backbone in the service